pkg/service/FluxApplicationService/FluxApplicationService.go

```diff
@@ -0,0 +1,87 @@
+package FluxApplicationService
+
+import (
+	"context"
+	k8sUtils "github.com/devtron-labs/common-lib/utils/k8s"
+	"github.com/devtron-labs/kubelink/converter"
+	client "github.com/devtron-labs/kubelink/grpc"
+	clusterRepository "github.com/devtron-labs/kubelink/pkg/cluster"
+	"go.uber.org/zap"
+	"k8s.io/client-go/rest"
+)
+
+type FluxApplicationService interface {
+	GetFluxApplicationListForCluster(config *client.ClusterConfig) *client.FluxApplicationList
+}
+
+type FluxApplicationServiceImpl struct {
+	logger            *zap.SugaredLogger
+	clusterRepository clusterRepository.ClusterRepository
+	k8sUtil           k8sUtils.K8sService
+	converter         converter.ClusterBeanConverter
+}
+
+func NewFluxApplicationServiceImpl(logger *zap.SugaredLogger,
+	clusterRepository clusterRepository.ClusterRepository,
+	k8sUtil k8sUtils.K8sService,
+	converter converter.ClusterBeanConverter) *FluxApplicationServiceImpl {
+	return &FluxApplicationServiceImpl{
+		logger:            logger,
+		clusterRepository: clusterRepository,
+		k8sUtil:           k8sUtil,
+		converter:         converter,
+	}
+
+}
+
+func (impl *FluxApplicationServiceImpl) GetFluxApplicationListForCluster(config *client.ClusterConfig) *client.FluxApplicationList {
+	impl.logger.Debugw("Fetching application list ", "clusterId", config.ClusterId, "clusterName", config.ClusterName)
+
+	appListFinal := make([]*FluxApplicationDto, 0)
+	k8sClusterConfig := impl.converter.GetClusterConfigFromClientBean(config)
+	restConfig, err := impl.k8sUtil.GetRestConfigByCluster(k8sClusterConfig)
+	if err != nil {
+		impl.logger.Errorw("Error in building rest config ", "clusterId", config.ClusterId, "err", err)
+		return &client.FluxApplicationList{}
+	}
+
+	var restConfig2 rest.Config
+	restConfig2 = *restConfig
```

Comment:
please add comment as to why are we copying the og rest handler value in another var, if possible also add a short note of the problem or link, so that other people are able to understand